data_structures/src/error.rs

```diff
@@ -55,16 +55,26 @@ pub enum TransactionError {
         change, expected_change
     )]
     InvalidTallyChange { change: u64, expected_change: u64 },
+    #[fail(
+        display = "Invalid witness reward found: {}. Expected value: {}",
+        value, expected_value
+    )]
+    InvalidReward { value: u64, expected_value: u64 },
+    #[fail(
+        display = "After complete a DataRequest live process, the total value in nwits is in invalid. Found: {}. Expected value: {}",
+        value, expected_value
+    )]
```

Comment:
Too cryptic of an error message.

Reply:
This is basically a sanity check in tally_validation, it should only happen if the implemented validation is wrong. It was created to ensure that there is no value created or burned. I wanted to use an assert but some people disagree.

How about:

    In tally validation, the total amount is incorrect. Found: {}. Expected value: {}